app/scripts/migrations/121.1.ts

```diff
@@ -0,0 +1,87 @@
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+import { CHAIN_IDS } from '../../../shared/constants/network';
+
+export const version = 121.1;
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+/**
+ * Migrates MATIC ticker in Network Configuration to POL ticker as per the direction in https://polygon.technology/blog/save-the-date-matic-pol-migration-coming-september-4th-everything-you-need-to-know
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>): void {
+  const networkControllerState = state.NetworkController;
+  if (
+    hasProperty(state, 'NetworkController') &&
+    isObject(networkControllerState) &&
+    hasProperty(networkControllerState, 'networkConfigurations') &&
+    isObject(networkControllerState.networkConfigurations)
+  ) {
+    for (const networkConfiguration of Object.values(
+      networkControllerState.networkConfigurations,
+    )) {
+      if (
+        isObject(networkConfiguration) &&
+        networkConfiguration.chainId === CHAIN_IDS.POLYGON
+      ) {
+        // update image path regardless of ticker
+        if (
+          hasProperty(networkConfiguration, 'rpcPrefs') &&
+          isObject(networkConfiguration.rpcPrefs) &&
+          hasProperty(networkConfiguration.rpcPrefs, 'imageUrl')
+        ) {
+          networkConfiguration.rpcPrefs.imageUrl = './images/pol-token.svg';
```

Comment:
Huh, why is there an `imageUrl` in the `rpcPrefs` state 🤔 This is missing from the `NetworkController` state type.

Reply:
It's there, so we'll need to migrate it, but I think we should follow up to remove this state completely. It looks like we don't need it, and it's introducing a type error (that will cause problems as we make progress with the TypeScript migration).

It looks like it is only in state because it leaked here from a constant. But it's not set dynamically, so there is no need for it to be in state. We can refactor places where this state is used to reference a constant instead. Or we can add it to the NetworkController type, if we want to start setting it dynamically as well.

Reply:
Sounds good. I may follow up with the Assets team here since I know there are a lot of NetworkController upgrades in flight.

Reply:
I have looked into this as part of my network controller upgrade and think I know what's going on.  Images were never supported in the network controller state, but leaked in somehow.  In my PR I move everything to grab them from constants.  We never supported a way for users to provide custom network images.

Reply:
Sounds like @bergeron has already dealt with this on the `NetworkController` upgrade work being done

Reply:
Yes, but to be clear it likely still needs to be migrated as of whichever 12.1.x this goes into.  So I'd keep the code here.